From fc927d8f1a6d81e5bcf58096871684b35b793fe2 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@strace.io>
Date: Wed, 27 Nov 2024 20:00:00 +0000
Subject: [PATCH] pam_access: fix group name match regression

* modules/pam_access/pam_access.c (group_match): Fix the order
of arguments passed to group_name_or_gid_match.

Resolves: https://github.com/linux-pam/linux-pam/issues/860
---
 modules/pam_access/pam_access.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/pam_access/pam_access.c b/modules/pam_access/pam_access.c
index 109115e9..15acbf94 100644
--- a/modules/pam_access/pam_access.c
+++ b/modules/pam_access/pam_access.c
@@ -766,7 +766,7 @@ group_match (pam_handle_t *pamh, char *tok, const char* usr, int debug)
     tok++;
     tok[strlen(tok) - 1] = '\0';
 
-    if (group_name_or_gid_match (pamh, usr, tok, debug))
+    if (group_name_or_gid_match (pamh, tok, usr, debug))
         return YES;
 
     return NO;
-- 
2.34.1

